### ISSUE TITLE
TypeError: Cannot read property 'append' of undefined after upgrading to 3.0.0

### ISSUE THREAD

**Issue by drzamich** [Hargne/jest-html-reporter#96]
I am using the `jest-html-reporter` to parse results of unit tests on that I run using command
```
CI=true react-scripts test  --silent --coverage --noStackTrace --ci --json --testResultsProcessor=./node_modules/jest-html-reporter --outputFile=./coverage/test-results.json"
```
It worked fine when I was using version `2.7.0`, when I upgraded to `3.0.0` I started getting such an error
```
TypeError: Cannot read property 'append' of undefined
    at h.setupConfig (/Users/mdrzazga/Work/mstrio_ui/production/node_modules/jest-html-reporter/dist/index.js:1:6242)
    at new h (/Users/mdrzazga/Work/mstrio_ui/production/node_modules/jest-html-reporter/dist/index.js:1:1341)
    at f (/Users/mdrzazga/Work/mstrio_ui/production/node_modules/jest-html-reporter/dist/index.js:1:8783)
    at module.exports (/Users/mdrzazga/Work/mstrio_ui/production/node_modules/jest-html-reporter/dist/index.js:1:8911)
    at processResults (/Users/mdrzazga/Work/mstrio_ui/production/node_modules/@jest/core/build/runJest.js:242:47)
    at /Users/mdrzazga/Work/mstrio_ui/production/node_modules/@jest/core/build/runJest.js:429:14
    at Generator.next (<anonymous>)
    at asyncGeneratorStep (/Users/mdrzazga/Work/mstrio_ui/production/node_modules/@jest/core/build/runJest.js:148:24)
    at _next (/Users/mdrzazga/Work/mstrio_ui/production/node_modules/@jest/core/build/runJest.js:168:9)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
error Command failed with exit code 1.
```
When I reverted back to `2.7.0` it was working again.
My `package.json`:
```json
{
  "name": "project",
  "version": "0.0.1",
  "private": true,
  "homepage": ".",
  "dependencies": {
    "antd": "3.19.1",
    "axios": "0.19.2",
    "i18next": "19.3.4",
    "line-clamp": "1.0.0",
    "moment": "2.24.0",
    "node-sass": "4.13.1",
    "react": "16.13.1",
    "react-dom": "16.13.1",
    "react-i18next": "11.3.4",
    "react-scripts": "3.4.1",
    "set-value": "3.0.1",
    "tabbable": "4.0.0"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "serve": "serve -s build",
    "test": "react-scripts test --env=jsdom",
    "test:all": "react-scripts test --env=jsdom --watchAll=false",
    "test:ci": "CI=true react-scripts test  --silent --noStackTrace --ci",
    "test:ci_report": "CI=true react-scripts test  --silent --coverage --noStackTrace --ci --json --testResultsProcessor=./node_modules/jest-html-reporter --outputFile=./coverage/test-results.json",
    "coverage": "react-scripts test --env=jsdom --coverage",
    "coverage:all": "react-scripts test --env=jsdom --coverage --watchAll=false",
    "eject": "react-scripts eject",
    "storybook": "start-storybook -p 9009 -s public",
    "storybook:build": "build-storybook -s public",
    "eslint": "eslint ./src --ext .js,.jsx",
    "eslint:report": "eslint ./src --ext .js,.jsx -f html -o ./eslint/index.html"
  },
  "devDependencies": {
    "@storybook/addon-knobs": "5.3.18",
    "@storybook/preset-create-react-app": "2.1.1",
    "@storybook/react": "5.3.18",
    "@storybook/theming": "5.3.18",
    "enzyme": "3.11.0",
    "enzyme-adapter-react-16": "1.15.2",
    "eslint": "6.8.0",
    "eslint-config-airbnb": "18.1.0",
    "eslint-plugin-import": "2.20.2",
    "eslint-plugin-jest": "23.8.2",
    "eslint-plugin-jsx-a11y": "6.2.3",
    "eslint-plugin-react": "7.19.0",
    "eslint-plugin-react-hooks": "3.0.0",
    "jest-html-reporter": "3.0.0",
    "prop-types": "15.7.2",
    "regenerator-runtime": "0.13.5"
  },
  "jest": {
    "collectCoverageFrom": [
      "src/utilities/**/*.{js,jsx}",
      "src/components/**/*.{js,jsx}",
      "!src/components/**/*.stories.{js,jsx}"
    ],
    "transformIgnorePatterns": [
      "/node_modules/(?!(react-dnd))/"
    ]
  },
  "browserslist": [
    ">0.2%",
    "not dead",
    "not ie < 11",
    "not op_mini all"
  ]
}
```

**Comment by Hargne** [comment]
Thanks for the report. I will look into this as soon as possible and hopefully release a fix for this. It would be awesome if you could help me verify it afterwards 👍

**Comment by drzamich** [comment]
Sure thing!

**Comment by Hargne** [comment]
I have now released v3.0.1-beta.0 with a fix for this issue. It would be awesome if you are able to confirm if this fixes your issue or not, and post the result in this thread 🙏 

#### Install by running

```
$ yarn add jest-html-reporter@3.0.1-beta.0
```

**Comment by drzamich** [comment]
I installed `3.0.1-beta.0` but now I am getting a different error
```
jest-html-reporter >> TypeError: Invalid date
jest-html-reporter >> TypeError: Cannot read property 'toString' of undefined
```

**Comment by Hargne** [comment]
Thanks for your report!
I will write a proper test case replicating your setup to avoid more issues like these.

**Comment by Hargne** [comment]
@drzamich

I have attempted to recreate your issue by setting up a similar setup as yours, but to no avail (or at least I am not getting the same errors). What I could find though was that there are a few quirks while using **create-react-app** together with Jest which might impact how this reporter is being handled.
First and foremost I would love it if you could install the latest beta **3.1.0-beta.0** which aims to fix the latest stated issue:

#### Install by running

```
$ yarn add jest-html-reporter@3.1.0-beta.0
```
*Please note that it is 3.1.0 and not 3.0.1*

But I would also recommend you to try to use the `reporters` option instead testResultsProcessor like so:

`--reporters=default --reporters=./node_modules/jest-html-reporter`

This will make the reporter run in parallel with Jest instead of afterwards and will therefore allow for a bit more control over the results.

Please report back the results 👍

**Comment by drzamich** [comment]
@Hargne 

after updating to `3.1.0-beta.0` the file was generated but with no content. So I followed your advice and edited the script and now it all works perfectly 👌! Thank you very much for quick response and professional assistance.

**Comment by Hargne** [comment]
Thanks for your help with resolving this issue! 🙏 
I will include this fix in the coming [v3.1.0](https://github.com/Hargne/jest-html-reporter/pull/100) which is currently in PR and will hopefully be released this week.
